plugin/trino-memsql/src/test/java/io/trino/plugin/memsql/TestMemSqlTypeMapping.java

```diff
@@ -582,6 +583,50 @@ private DataTypeTest dateTestCases(DataType<LocalDate> dateDataType, ZoneId jvmZ
                 .addRoundTrip(dateDataType, dateOfLocalTimeChangeBackwardAtMidnightInSomeZone);
     }
 
+    @Test(dataProvider = "sessionZonesDataProvider")
+    public void testTime(ZoneId sessionZone)
+    {
+        Session session = Session.builder(getSession())
+                .setTimeZoneKey(getTimeZoneKey(sessionZone.getId()))
+                .build();
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("time(0)", "NULL", TIME_SECONDS, "CAST(NULL AS time(0))")
+                .addRoundTrip("time(0)", "TIME '00:00:00'", TIME_SECONDS, "TIME '00:00:00'")
+                .addRoundTrip("time(0)", "TIME '01:02:03'", TIME_SECONDS, "TIME '01:02:03'")
+                .addRoundTrip("time(0)", "TIME '23:59:59'", TIME_SECONDS, "TIME '23:59:59'")
+                .execute(getQueryRunner(), session, trinoCreateAsSelect("tpch.test_time"))
+                .execute(getQueryRunner(), session, trinoCreateAndInsert(getSession(), "tpch.test_time"));
+
+        SqlDataTypeTest.create()
+                .addRoundTrip("time", "NULL", TIME_SECONDS, "CAST(NULL AS time(0))")
+                .addRoundTrip("time", "'00:00:00'", TIME_SECONDS, "TIME '00:00:00'")
+                .addRoundTrip("time", "'01:02:03'", TIME_SECONDS, "TIME '01:02:03'")
+                .addRoundTrip("time", "'23:59:59'", TIME_SECONDS, "TIME '23:59:59'")
+                .execute(getQueryRunner(), session, memSqlCreateAndInsert("tpch.test_time"));
+    }
+
+    @Test(dataProvider = "unsupportedTimeDataProvider")
+    public void testUnsupportedTime(String unsupportedTime)
+    {
+        try (TestTable table = new TestTable(memSqlServer::execute, "tpch.test_unsupported_time", "(col time)", ImmutableList.of(format("'%s'", unsupportedTime)))) {
+            assertQueryFails(
+                    "SELECT * FROM " + table.getName(),
+                    format("\\Q%s cannot be parse as LocalTime (format is \"HH:mm:ss[.S]\" for data type \"TIME\")", unsupportedTime));
+        }
+    }
+
+    @DataProvider
+    public Object[][] unsupportedTimeDataProvider()
```

Comment:
Maybe add a link to the memsql docs on the `time` data type:
https://docs.singlestore.com/db/v7.6/en/reference/sql-reference/data-types/time-and-date.html?action=version-change

Additionally, what happens with a time that has more precision than second (I'm assuming an error of some sort)? Is it worth adding a testcase for that?

Reply:
@grantatspothero Link to official data types will be added in another PR #10296. 

Also, added some testcases.